sw/device/lib/base/bitfield.h

```diff
@@ -0,0 +1,34 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_BASE_BITFIELD_H_
+#define OPENTITAN_SW_DEVICE_LIB_BASE_BITFIELD_H_
+
+#include <stdint.h>
+
+/**
+ * Masked field within a 32 bit bitfield.
```

Comment:
Like in #1787, I would like to see a lot more documentation for this struct.

Reply:
When #1787 is approved, I will copy paste the description from there.

Reply:
Done.